test/extended/openstack/local-bd-etcd.go

```diff
@@ -0,0 +1,158 @@
+package openstack
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/gophercloud/gophercloud"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/stretchr/objx"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+	e2eskipper "k8s.io/kubernetes/test/e2e/framework/skipper"
+)
+
+var _ = g.Describe("[sig-installer][Suite:openshift/openstack] The OpenShift cluster", func() {
+	defer g.GinkgoRecover()
+
+	const minEtcdDiskSizeGiB = 7
+	const etcdBlockDeviceName = "etcd"
+
+	var computeClient *gophercloud.ServiceClient
+	var ctx context.Context
+	var dc dynamic.Interface
+	var controlPlaneFlavor string
+	var clusterCPMS objx.Map
+	var clientSet *kubernetes.Clientset
+
+	checkEtcdDisk := func(machine objx.Map) error {
+		additionalBlockDevices := machine.Get("spec.providerSpec.value.additionalBlockDevices").ObjxMapSlice()
+		if len(additionalBlockDevices) == 0 {
+			return fmt.Errorf("machine %q does not have additional block devices", machine.Get("metadata.name").String())
+		}
+
+		for _, blockDevice := range additionalBlockDevices {
+			if blockDevice.Get("name").String() == etcdBlockDeviceName {
+				if blockDevice.Get("sizeGiB").Int() < minEtcdDiskSizeGiB {
+					return fmt.Errorf("machine %q has an additional block device with size %d GiB, which is less than the minimum required %d GiB", machine.Get("metadata.name").String(), blockDevice.Get("sizeGiB").Int(), minEtcdDiskSizeGiB)
+				}
+				if blockDevice.Get("storage.type").String() != "Local" {
+					return fmt.Errorf("machine %q has an additional block device with storage type %q, which is not Local", machine.Get("metadata.name").String(), blockDevice.Get("storage.type").String())
+				}
+				return nil
+			}
+		}
+		return fmt.Errorf("machine %q does not have an additional block device named %s", machine.Get("metadata.name").String(), etcdBlockDeviceName)
+	}
+
+	// This function will be removed https://github.com/openshift/openstack-test/pull/127 is merged.
+	getControlPlaneMachines := func(ctx context.Context, dc dynamic.Interface) ([]objx.Map, error) {
+		controlPlaneMachines := make([]objx.Map, 0)
+		mc := dc.Resource(schema.GroupVersionResource{
+			Group:    machineAPIGroup,
+			Version:  "v1",
+			Resource: "machines",
+		}).Namespace(machineAPINamespace)
+		obj, err := mc.List(ctx, metav1.ListOptions{})
+		if err != nil {
+			return nil, err
+		}
+		for _, m := range objects(objx.Map(obj.UnstructuredContent()).Get("items")) {
+			if m.Get("machine.openshift.io/cluster-api-machine-role").String() == "master" {
```

Comment:
I am not sure this is correct. At a minimum, shouldn't you look into a `labels` property?

Anyway, you can start using https://github.com/openshift/openstack-test/pull/127 as soon as we merge it (and if I got it wrong, then we will collectively notice & fix!)

Reply:
yeah I'll refactor on your PR, yours will merge today I think. I'm waiting for CI and I'll approve.
Thanks for your work!

Reply:
#127 has merged, should we add the changes Pierre asked for in https://github.com/openshift/openstack-test/pull/128 ?

Reply:
that's what I did in the other PR? I think your comment it outdated